ReSharper.FSharp/src/FSharp.Psi.Features/src/Util/FSharpExpressionUtil.fs

```diff
@@ -12,6 +12,26 @@ open JetBrains.ReSharper.Psi
 open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
 open JetBrains.ReSharper.Psi.Tree
 
+/// If given an expression is a return expression, the owning IBinding is returned, otherwise null.
+let getReturnExpressionOwner (expr: ISynExpr) : IBinding =
+    let rec getParentFunction (treeNode : ITreeNode) =
+        let parent = treeNode.Parent
```

Comment:
I'd probably expect it to be `ISynExpr -> ISynExpr` (though, `ISynExpr` was recently renamed to `IFSharpExpression` after the FCS-inspired name produced many questions).

Getting an outer expression that is a parent to some return expression is needed in various places and is not limited to `let`-bindings. Speaking of top-level declarations, it can also be used for auto properties or other members in types. Other cases, for instance, include case where it'd help to reason about a type of an `if` expression by looking at its return expression in `if` branch.

This code should really cover common cases via generated navigators like:
`IfThenElseExprNavigator.GetByBranchExpression`
`MatchClauseNavigator.GetByExpression`
`MatchLikeExprNavigator.GetByClause`
and so on.

In most cases simply taking a parent is wrong (and this is why it wasn't written this way with a `todo` to cover it properly):
```fsharp
match 123 with
| x when { } -> { }
```
In the code above there are multiple expressions including two unfinished record ones. Only the second of this two is considered a "return" expression for the match expression.